command/setup_consul.go

```diff
@@ -0,0 +1,485 @@
+	s.Ui.Output(fmt.Sprintf(authMethodMsg, authMethodServices, authMethodTasks))
+
+	authMethodConf, err := s.renderAuthMethodConf()
+	if err != nil {
+		s.Ui.Error(err.Error())
+		return 1
+	}
+
+	jsConf, _ := json.MarshalIndent(authMethodConf, "", "    ")
+	s.Ui.Output(string(jsConf))
+
+	if s.consulEnt {
+		namespaceMsg := `
+Since you're running Consul Enterprise, we will additionally create
+a namespace %s and bind the auth methods to that namespace.
+`
+		s.Ui.Output(fmt.Sprintf(namespaceMsg, consulNamespace))
+	}
+
+	var createAuthMethods bool
+	if !s.autoYes {
+		createAuthMethods = s.askQuestion("Create these auth methods in your Consul cluster? [Y/n]")
+	} else {
+		createAuthMethods = true
+	}
+
+	if s.consulEnt {
+		err = s.createNamespace()
+		if err != nil {
+			s.Ui.Error(err.Error())
+			return 1
+		}
+	}
+
+	if createAuthMethods {
+		err = s.createAuthMethod(authMethodServices, authMethodConf)
+		if err != nil {
+			s.Ui.Error(err.Error())
+			return 1
+		}
+		err = s.createAuthMethod(authMethodTasks, authMethodConf)
+		if err != nil {
+			s.Ui.Error(err.Error())
+			return 1
+		}
+	}
+
+	servicesBindingRule := &api.ACLBindingRule{
+		Description: "binding rule for Nomad services w/ (WI)",
+		AuthMethod:  authMethodServices,
+		BindType:    "service",
+		BindName:    "${value.nomad_namespace}-${value.nomad_service}",
+	}
+
+	tasksBindingRule := &api.ACLBindingRule{
+		Description: "binding rule for Nomad tasks w/ (WI)",
+		AuthMethod:  authMethodTasks,
+		BindType:    "role",
+		BindName:    "nomad-${value.nomad_namespace}-templates",
+	}
+
+	s.Ui.Output(`
+In order to map claims between Nomad's JWTs and Consul ACL, we need to create
+the following binding rules:
+`)
+	jsServicesBindingRule, _ := json.MarshalIndent(servicesBindingRule, "", "    ")
+	jsTasksBindingRule, _ := json.MarshalIndent(tasksBindingRule, "", "    ")
+	s.Ui.Output(string(jsServicesBindingRule))
+	s.Ui.Output(string(jsTasksBindingRule))
+
+	var createBindingRules bool
+	if !s.autoYes {
+		createBindingRules = s.askQuestion(
+			"Create these binding rules in your Consul cluster? [Y/n]",
+		)
+	} else {
+		createBindingRules = true
+	}
+
+	if createBindingRules {
+		err = s.createBindingRules(servicesBindingRule)
+		if err != nil {
+			s.Ui.Error(err.Error())
+			return 1
+		}
+		err = s.createBindingRules(tasksBindingRule)
+		if err != nil {
+			s.Ui.Error(err.Error())
+			return 1
+		}
+	}
+
+	s.Ui.Output(`
+Nomad tasks require a Consul ACL policy and Role. Below is the body of the policy
+we need to create:
+`)
+	s.Ui.Output(string(policyBody))
+
+	var createPolicy bool
+	if !s.autoYes {
+		createPolicy = s.askQuestion(
+			"Should we create the above policy in your Consul cluster? [Y/n]",
+		)
+	} else {
+		createPolicy = true
+	}
+
+	if createPolicy {
+		err = s.createPolicy()
+		if err != nil {
+			s.Ui.Error(err.Error())
+			return 1
+		}
+	}
+
+	s.Ui.Output(`
+Finally, we need to create a role role-nomad-tasks associated with the policy
+above.`)
+
```

Comment:
I think we're missing printing the role information.

Reply:
a role is just a role name we print above and a link to a policy.